it/src/main/java/com/google/cloud/teleport/it/bigquery/DefaultBigQueryResourceManager.java

```diff
@@ -0,0 +1,369 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.bigquery;
+
+import static com.google.cloud.teleport.it.bigquery.BigQueryResourceManagerUtils.checkValidTableId;
+
+import com.google.api.gax.paging.Page;
+import com.google.auth.Credentials;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryError;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetInfo;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.InsertAllRequest.RowToInsert;
+import com.google.cloud.bigquery.InsertAllResponse;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDefinition;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default class for implementation of {@link BigQueryResourceManager} interface.
+ *
+ * <p>The class supports one dataset, and multiple tables per dataset object.
+ *
+ * <p>The class is thread-safe.
+ */
+public final class DefaultBigQueryResourceManager implements BigQueryResourceManager {
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultBigQueryResourceManager.class);
+  private static final String DEFAULT_DATASET_REGION = "us-central1";
+
+  private final String testId;
+  private final String projectId;
+  private final String datasetId;
+
+  private final BigQuery bigQuery;
+  private Dataset dataset;
+
+  private DefaultBigQueryResourceManager(Builder builder, BigQuery bigQuery) {
+    this.testId = builder.getTestId();
+    this.projectId = builder.getProjectId();
+    this.datasetId = builder.getDatasetId();
+    this.bigQuery = bigQuery;
+  }
+
+  public static DefaultBigQueryResourceManager.Builder builder(String testId, String projectId) {
+    return new DefaultBigQueryResourceManager.Builder(testId, projectId);
+  }
+
+  /**
+   * Returns the project ID this Resource Manager is configured to operate on.
+   *
+   * @return the project ID.
+   */
+  public String getProjectId() {
+    return projectId;
+  }
+
+  /**
+   * Return the dataset ID this Resource Manager uses to create and manage tables in.
+   *
+   * @return the dataset ID.
+   */
+  public String getDatasetId() {
+    return datasetId;
+  }
+
+  /**
+   * Return the test ID this Resource Manager uses to generate the instance and cluster(s) ID's
+   * from.
+   *
+   * @return test ID.
+   */
+  public String getTestId() {
```

Comment:
Similar to BigtableResourceManager in PR #445: no need to store and expose `testId`?

Reply:
Agreed. I removed all instances and tests for testId.